src/Equinox.MessageDb/MessageDbClient.fs

```diff
@@ -32,29 +42,31 @@ module private Npgsql =
 
 type MessageDbWriter(connectionString : string) =
 
-    let prepareAppend (streamName : string) (expectedVersion : int64) (e : IEventData<Format>) =
-        let cmd = NpgsqlBatchCommand(CommandText = "select 1 from write_message(@Id::text, @StreamName, @EventType, @Data, @Meta, @ExpectedVersion)")
+    let prepareAppend (streamName : string) (expectedVersion : int64 option) (e : IEventData<Format>) =
+        let cmd = NpgsqlBatchCommand(CommandText = "select * from write_message(@Id::text, @StreamName, @EventType, @Data, @Meta, @ExpectedVersion)")
 
         cmd.Parameters.AddWithValue("Id", NpgsqlDbType.Uuid, e.EventId) |> ignore
         cmd.Parameters.AddWithValue("StreamName", NpgsqlDbType.Text, streamName) |> ignore
         cmd.Parameters.AddWithValue("EventType", NpgsqlDbType.Text, e.EventType) |> ignore
         cmd.Parameters |> Json.addParameter "Data" e.Data
         cmd.Parameters |> Json.addParameter "Meta" e.Meta
-        cmd.Parameters.AddWithValue("ExpectedVersion", NpgsqlDbType.Bigint, expectedVersion) |> ignore
+        cmd.Parameters |> Sql.addNullableInt64 "ExpectedVersion" expectedVersion
 
         cmd
 
-    member _.WriteMessages(streamName, events : _ array, version : int64, ct) = task {
+    member _.WriteMessages(streamName, events : _ array, ct, ?version : int64) = task {
```

Comment:
I hate this api, version shouldn't be an optional arg, it's an option

Reply:
yeah, should ahve trapped earlier